gtsam/navigation/ConstantVelocityFactor.h

```diff
@@ -0,0 +1,53 @@
+#include <gtsam/base/numericalDerivative.h>
+#include <gtsam/geometry/Pose3.h>
+#include <gtsam/inference/Symbol.h>
+#include <gtsam/navigation/NavState.h>
+#include <gtsam/nonlinear/NonlinearFactor.h>
+
+namespace gtsam {
+
+class ConstantVelocityFactor : public NoiseModelFactor2<NavState, NavState> {
+    double dt_;
+
+   public:
+    using Base = NoiseModelFactor2<NavState, NavState>;
+
+   public:
+    ConstantVelocityFactor(Key i, Key j, double dt, const SharedNoiseModel &model)
+        : NoiseModelFactor2<NavState, NavState>(model, i, j), dt_(dt) {}
+    ~ConstantVelocityFactor() override{};
+
+    gtsam::Vector evaluateError(const NavState &x1, const NavState &x2,
+                                boost::optional<gtsam::Matrix &> H1 = boost::none,
+                                boost::optional<gtsam::Matrix &> H2 = boost::none) const override {
+        // so we can reuse this calculation below
+        auto evaluateErrorInner = [dt = dt_](const NavState &x1, const NavState &x2) -> gtsam::Vector {
+            const Velocity3 &v1 = x1.v();
+            const Velocity3 &v2 = x2.v();
+            const Point3 &p1 = x1.t();
+            const Point3 &p2 = x2.t();
+
+            // trapezoidal integration constant accelleration
+            // const Point3 hx = p1 + Point3((v1 + v2) * dt / 2.0);
+
+            // euler start
+            // const Point3 hx = p1 + Point3(v1 * dt);
+
+            // euler end
+            const Point3 hx = p1 + Point3(v2 * dt);
+
+            return p2 - hx;
+        };
+
+        if (H1) {
+            (*H1) = numericalDerivative21<gtsam::Vector, NavState, NavState>(evaluateErrorInner, x1, x2, 1e-5);
```

Comment:
I think what is needed here is a call to

```
Vector9 NavState::localCoordinates(const NavState& g, OptionalJacobian<9, 9> H1, OptionalJacobian<9, 9> H2) const {
```

where you evaluate the difference between `predicted` and `x2`, as in:

```
NavState predicted = (something that yields predicted_H_x1)
Vector9 error = predicted.localCoordinates(x2, error_H_predicted, H2);
if (H1) *H1 = error_H_predicted * predicted_H_x1;
```

Reply:
Any reason I can't just leverage the navstate.update(body_accel,body_omega, dt) function here to get my predicted NavState? I will want to use those terms later (and build out a NavState++ that can store those states as well) so this would get me closer to that final goal.

Reply:
I took a pass at this style. Seems to behave, but let me know if I missunderstood something. Obviously I need to test the rotation cases and find a cleaner way to init a Vector9 to clean up the verbosity of the tests.